### ISSUE TITLE
loadups, pseudo hosts, loadup version stamp?

### ISSUE THREAD

**Issue by masinter** [Interlisp/medley#1217]
Maybe this should be a Discussion?

Can we make Pseudo hosts like {MEDLEY} reliable enough that we could use it instead of MEDLEYDIR?
Shuld the loadups show {MEDLEY}<lisp> and {MEDLEY}<lispusers> in LOADEDFILELST? Do you keep the version numbers? You're more likely to see version number changes when the file systemss change -- i think the evnironment might give more validity to versioned file names than is warrented.

Once I brought up possibly moving away from {DSK} and got some push-back, but what I meant was keeping {UNIX} (or {} or {HOST} to be the posix file system, and implement all of the ~N~ veersion stuff in Lisp. 
Basically, {DSK} would be a (versioning, case insensitive) kind of pseudo-host.

I wonder if {HOME} might be better than {IL} (@fghalasz) or {LI} (@rmkaplan ) or even {LOGIN} or {HOME} (for $#r foe youe homw dir).

We could do this one change at a time, but they seem interrelated enough to try to plan.

making pseudohosts two directional pick the shortest form among those registered?

**Comment by rmkaplan** [comment]
FWIW, I have been running for months with no gliches in a strictly pseudohost world ( (MEDLEY}, my working {WMEDLEY}, {TEST}, {OT} (old Tedit files, for me), {RELEASE} , {LFG}…).

GITFNS sets up some of them relative to MEDLEYDIR, my INIT sets up the other ones.   I only see pseudohosts on DIRECTORIES, LISPUSERSDIRECTORIES, FONTDIRECTORIES, etc.

Of course, this doesn’t deal with the fact that version numbers are different for {MEDLEY} files on different machines.  Version numbers are valuable locally (I have hundreds of versions of some of the Tedit files), and I have little tools for looking at particular definitions from particular files, for when I get ahead of myself. 

I’m not sure what you mean by “two directional”.  Pseudohost has functions that map file names either way, and they could be combined to get the shortest, if that’s what you are after.  But I think the pseudohost name will almost always win (unless Medley is really at the top-level of your file system).

**Comment by nbriggs** [comment]
`{DSK}` is expected to be the versioned local file system, `{UNIX}` the un-versioned raw Unix file system.  One or other of them is what underpins *local* references via a pseudo-host, and the particular underlying file system the pseudo-host is pointing to should define the semantics (case-sensitivity, versioning, character-set of the names?) of directory/file names on that pseudo-host. (e.g., files on NS file servers could have NS characters in their names... files on an IFS couldn't [as far as I recall])

**Comment by nbriggs** [comment]
We have to keep in mind that we don't want to end up with a dependency on a pseudo-host named file before the pseudo-host implementation is loaded (because we rewrote an early file from a fully-loaded system, say)- it might not be too difficult, since the system probably had to ignore non-existent network hosts and re-locate the file file based on the base name and the DIRECTORIES setting anyway, but worth thinking about when making changes in this area.

**Comment by rmkaplan** [comment]
I think a pseudohost has the same properties as the host it is defined on, at least that’s the way it is supposed to work.  You could have a pseudohost on {UNIX}/foo/  and a different pseudohost on {DSK}/foo/ and you would have to access the same underlying files in different ways (e.g. xxx.~25~ vs  xxx.;25).  A pseudo-filedevice basically just forwards its methods to the underlying filedevice.
